src/openvic-simulation/modifier/ModifierManager.cpp

```diff
@@ -686,181 +674,105 @@ bool ModifierManager::parse_scripts(DefinitionManager const& definition_manager)
-
-		if (dryad::node_has_kind<ast::IdentifierValue>(value)) {
-			ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
-
-			if (effect != nullptr) {
-				return add_modifier_cb(effect, value);
-			} else if (key == "war_exhaustion_effect") {
-				Logger::warning("war_exhaustion_effect does nothing (vanilla issues have it).");
-				return true;
-			} else {
-				const decltype(modifier_effect_variants)::const_iterator effect_it = modifier_effect_variants.find(key);
-
-				if (effect_it != modifier_effect_variants.end()) {
-					effect_variant_map_t const& variants = effect_it->second;
-
-					const effect_variant_map_t::const_iterator variant_it = variants.find(type);
-
-					if (variant_it != variants.end()) {
-						effect = variant_it->second;
-
-						if (effect != nullptr) {
-							return add_modifier_cb(effect, value);
-						}
-					}
-
-					Logger::error(
-						"Modifier effect \"", key, "\" does not have a valid variant for use in ",
-						Modifier::modifier_type_to_string(type), " modifiers."
-					);
-					return false;
-				}
-			}
-		} else if (dryad::node_has_kind<ast::ListValue>(value) && complex_modifiers.contains(key)) {
-			if (key == "rebel_org_gain") { // because of course there's a special one
-				std::string_view faction_identifier;
-				ast::NodeCPtr value_node = nullptr;
-
-				bool ret = expect_dictionary_keys(
-					"faction", ONE_EXACTLY, expect_identifier(assign_variable_callback(faction_identifier)),
-					"value", ONE_EXACTLY, assign_variable_callback(value_node)
-				)(value);
-
-				ret &= add_flattened_modifier_cb(key, faction_identifier, value_node);
-
-				return ret;
-			} else {
-				return expect_dictionary(std::bind_front(add_flattened_modifier_cb, key))(value);
-			}
+NodeTools::key_value_callback_t ModifierManager::_expect_modifier_effect(
+	CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque> const& registry,
+	ModifierValue& modifier_value
+) const {
+	return _expect_modifier_effect_with_fallback(registry, modifier_value, key_value_invalid_callback);
+}
+NodeTools::key_value_callback_t ModifierManager::_expect_modifier_effect_with_fallback(
+	CaseInsensitiveIdentifierRegistry<ModifierEffect, RegistryStorageInfoDeque> const& registry,
+	ModifierValue& modifier_value,
+	NodeTools::key_value_callback_t const& fallback
+) const {
+	return [this, &registry, &modifier_value, fallback = FWD(fallback)](auto const& key, auto const& value) {
+		if (dryad::node_has_kind<ast::ListValue>(value) && complex_modifiers.contains(key)) {
+			const key_value_callback_t callback = [this, &modifier_value, &key](auto const& inner_key, auto const& value) -> bool{
+				return _add_flattened_modifier_cb(modifier_value, key, inner_key, value);
+			};
+			return expect_dictionary(callback)(value);
 		}
-
-		return default_callback(key, value);
+		
+		ModifierEffect const* effect = registry.get_item_by_identifier(key);
+		if(effect == nullptr) { return fallback(key, value); }
```

Comment:
```suggestion
		if (effect == nullptr) {
			return fallback(key, value);
		}
```
Formatting standards - space after `if`, never compress expressions like this onto a single line